injectproxy/rules.go

```diff
@@ -223,9 +223,51 @@ func (r *routes) filterRules(lvalues []string, resp *apiResponse) (interface{},
 	filtered := []*ruleGroup{}
 	for _, rg := range rgs.RuleGroups {
 		var rules []rule
-		for _, rule := range rg.Rules {
-			if lval := rule.Labels().Get(r.label); lval != "" && m.Matches(lval) {
-				rules = append(rules, rule)
+		for _, rgr := range rg.Rules {
+			if lval := rgr.Labels().Get(r.label); lval != "" && m.Matches(lval) {
+				rules = append(rules, rgr)
+				continue
+			}
+
+			if !r.rulesWithActiveAlerts || rgr.alertingRule == nil {
+				continue
+			}
+
+			var ar *alertingRule
+			for i := range rgr.alertingRule.Alerts {
+				if lval := rgr.alertingRule.Alerts[i].Labels.Get(r.label); lval == "" || !m.Matches(lval) {
+					continue
+				}
+
+				if ar == nil {
+					ar = &alertingRule{
```

Comment:
I have one question here: Could you explain to me, why we have to copy the struct instead using 

```
ar = rgr.alertingRule 
```

I thought, after go 1.22, the loopvar issue is gone?

Reply:
Thanks for the clarification